src/components/shapes/draw.js

```diff
@@ -605,13 +607,30 @@ function drawLabel(gd, index, options, shapeGroup) {
     // Remove existing label
     shapeGroup.selectAll('.shape-label').remove();
 
-    // If no label, return
-    if(!options.label.text) return;
+    // If no label text or texttemplate, return
+    if(!(options.label.text || options.label.texttemplate)) return;
+
+    // Text template overrides text
+    var text = options.label.text;
+    if(options.label.texttemplate) {
+        var templateValues = {};
+        if(options.type !== 'path') {
```

Comment:
Makes sense.

I think we still need the check here though, because a texttemplate might get applied to a new path drawn by the user if `newshape.texttemplate` is set.

Or can we prevent the texttemplate from being applied to the new shape if the user draws a path?